src/cache/cache-manager.ts

```diff
@@ -20,6 +22,47 @@ export class CacheManager {
     this.nowProvider = this.nowProvider || DEFAULT_NOW_PROVIDER;
   }
 
+  async setIdToken(
+    clientId: string,
+    idToken: string,
+    decodedToken: DecodedToken
+  ): Promise<void> {
+    const cacheKey = this.getIdTokenCacheKey(clientId);
+    await this.cache.set(cacheKey, {
+      id_token: idToken,
+      decodedToken
+    });
+    await this.keyManifest?.add(cacheKey);
+  }
+
+  async getIdToken(
+    cacheKey: CacheKey
+  ): Promise<{ id_token: string; decodedToken: DecodedToken }> {
+    let entry = await this.cache.get<{
+      id_token: string;
+      decodedToken: DecodedToken;
+    }>(this.getIdTokenCacheKey(cacheKey.clientId));
+
+    if (!entry && cacheKey.scope && cacheKey.audience) {
+      const audience = cacheKey.audience;
+      const scope = cacheKey.scope;
+      const entryByScope = await this.get(
+        new CacheKey({
```

Comment:
I _think_ we're reconstructing the provided `cacheKey` here, could we reuse that instead of creating a new one?